web/src/main/java/org/springframework/security/web/access/RequestMatcherDelegatingWebInvocationPrivilegeEvaluator.java

```diff
@@ -0,0 +1,179 @@
+	 * matched, or if there is not an available {@code WebInvocationPrivilegeEvaluator},
+	 * returns {@code true}.
+	 * @param uri the URI excluding the context path (a default context path setting will
+	 * be used)
+	 * @return true if access is allowed, false if denied
+	 */
+	@Override
+	public boolean isAllowed(String uri, Authentication authentication) {
+		return isAllowed(null, uri, null, authentication);
+	}
+
+	/**
+	 * Determines whether the user represented by the supplied <tt>Authentication</tt>
+	 * object is allowed to invoke the supplied URI.
+	 * <p>
+	 * Uses the provided URI in the
+	 * {@link org.springframework.security.web.util.matcher.RequestMatcher#matches(HttpServletRequest)}
+	 * for every {@code RequestMatcher} configured. If no {@code RequestMatcher} is
+	 * matched, or if there is not an available {@code WebInvocationPrivilegeEvaluator},
+	 * returns {@code true}.
+	 * @param uri the URI excluding the context path (a default context path setting will
+	 * be used)
+	 * @param contextPath the context path (may be null, in which case a default value
+	 * will be used).
+	 * @param method the HTTP method (or null, for any method)
+	 * @param authentication the <tt>Authentication</tt> instance whose authorities should
+	 * be used in evaluation whether access should be granted.
+	 * @return true if access is allowed, false if denied
+	 */
+	@Override
+	public boolean isAllowed(String contextPath, String uri, String method, Authentication authentication) {
+		DelegatePrivilegeEvaluator delegate = getDelegate(contextPath, uri, method);
+		if (delegate == null) {
+			return true;
+		}
+		List<WebInvocationPrivilegeEvaluator> privilegeEvaluators = delegate.getPrivilegeEvaluators();
+		if (CollectionUtils.isEmpty(privilegeEvaluators)) {
+			return true;
+		}
+		for (WebInvocationPrivilegeEvaluator evaluator : privilegeEvaluators) {
+			boolean isAllowed = evaluator.isAllowed(contextPath, uri, method, authentication);
+			if (!isAllowed) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	public static Builder withRequestMatcher(Function<HttpServletRequest, Boolean> requestMatcher) {
+		DelegatePrivilegeEvaluator delegatePrivilegeEvaluator = new DelegatePrivilegeEvaluator(requestMatcher);
+		return new Builder(delegatePrivilegeEvaluator);
+	}
+
+	private DelegatePrivilegeEvaluator getDelegate(String contextPath, String uri, String method) {
+		FilterInvocation filterInvocation = new FilterInvocation(contextPath, uri, method);
+		for (DelegatePrivilegeEvaluator delegate : this.delegates) {
+			if (delegate.matches(filterInvocation.getHttpRequest())) {
+				return delegate;
+			}
+		}
+		return null;
+	}
+
+	public static class DelegatePrivilegeEvaluator {
+
+		private final Function<HttpServletRequest, Boolean> requestMatcher;
+
+		private final List<WebInvocationPrivilegeEvaluator> privilegeEvaluators;
+
+		public DelegatePrivilegeEvaluator(Function<HttpServletRequest, Boolean> requestMatcher) {
```

Comment:
The code should not allow one to be created without any WebInvocationProivilegeEvaluator instances because we don't want the `RequestMatcher` to be invoked unnecessarily (it decreases the performance every time a check is performed.